packages/editor/src/services/LockedState.ts

```diff
@@ -22,19 +22,31 @@ Original Code is the Infinite Reality Engine team.
 All portions of the code written by the Infinite Reality Engine team are Copyright © 2021-2023 
 Infinite Reality Engine. All Rights Reserved.
 */
-
-import { defineComponent, hasComponent, removeComponent, setComponent } from '@ir-engine/ecs/src/ComponentFunctions'
-import { Entity } from '@ir-engine/ecs/src/Entity'
-import { S } from '@ir-engine/ecs/src/schemas/JSONSchemas'
-
-export const LockedComponent = defineComponent({
-  name: 'LockedComponent',
-  jsonID: 'EE_locked',
-  schema: S.Bool(false)
+import { Entity } from '@ir-engine/ecs'
+import { defineState, getMutableState } from '@ir-engine/hyperflux'
+
+export const LockedState = defineState({
+  name: 'LockedState',
+  initial: () => ({
+    lockedEntities: new Map<Entity, boolean>() // Map to store locked state of entities
```

Comment:
this should be a record/object, otherwise it will not be reactive. Maps and Sets are not reactive in Hookstate.